healthcheck_test.go

```diff
@@ -196,3 +230,187 @@ func TestExceptionUnreachable(t *testing.T) {
 		})
 	})
 }
+
+func TestIndexExists(t *testing.T) {
+	Convey("Given that the client has one index and this index exists", t, func() {
+
+		httpCli := &dphttp.ClienterMock{DoFunc: func(ctx context.Context, request *http.Request) (*http.Response, error) {
+			if request.Method == "HEAD" {
+				return doIndexExists(ctx, request)
+			}
+			return doOkGreen(ctx, request)
+		}}
+
+		cli := elasticsearch.NewClientWithHTTPClient(testUrl, true, httpCli, testIndex)
+
+		// CheckState for test validation
+		checkState := health.NewCheckState(elasticsearch.ServiceName)
+
+		Convey("Checker updates the CheckState to a successful state", func() {
+			cli.Checker(context.Background(), checkState)
+			So(len(httpCli.DoCalls()), ShouldEqual, 2)
+			So(checkState.Status(), ShouldEqual, health.StatusOK)
+			So(checkState.Message(), ShouldEqual, elasticsearch.MsgHealthy)
+			So(checkState.StatusCode(), ShouldEqual, 200)
+
+		})
+	})
+}
+
+func TestIndexDoesNotExist(t *testing.T) {
+	Convey("Given that the client has one index and this index does not exists", t, func() {
+
+		httpCli := &dphttp.ClienterMock{DoFunc: func(ctx context.Context, request *http.Request) (*http.Response, error) {
+			if request.Method == "HEAD" {
+				return doIndexDoesNotExist(ctx, request)
+			}
+			return doOkGreen(ctx, request)
+		}}
+
+		cli := elasticsearch.NewClientWithHTTPClient(testUrl, true, httpCli, testIndex)
+
+		// CheckState for test validation
+		checkState := health.NewCheckState(elasticsearch.ServiceName)
+
+		Convey("Checker updates the CheckState to a successful state", func() {
+			cli.Checker(context.Background(), checkState)
+			So(len(httpCli.DoCalls()), ShouldEqual, 2)
+			So(checkState.Status(), ShouldEqual, health.StatusCritical)
+			So(checkState.Message(), ShouldEqual, elasticsearch.ErrorIndexDoesNotExist.Error())
+			So(checkState.StatusCode(), ShouldEqual, 404)
+
+		})
+	})
+}
+
+func TestNoClientIndex(t *testing.T) {
+	Convey("Given that the client has one index and this index does not exists", t, func() {
+
+		httpCli := &dphttp.ClienterMock{DoFunc: func(ctx context.Context, request *http.Request) (*http.Response, error) {
+			if request.Method == "HEAD" {
+				return doNoClientIndex(ctx, request)
+			}
+			return doOkGreen(ctx, request)
+		}}
+
+		cli := elasticsearch.NewClientWithHTTPClient(testUrl, true, httpCli, testNoClientIndex)
+
+		// CheckState for test validation
+		checkState := health.NewCheckState(elasticsearch.ServiceName)
+
+		Convey("Checker updates the CheckState to a successful state", func() {
+			cli.Checker(context.Background(), checkState)
+			So(len(httpCli.DoCalls()), ShouldEqual, 1)
```

Comment:
It would be a good idea to check the request in the DoCall, to validate that there was no call against an index endpoint? (This comment would apply to all the tests, to validate that the requests are what we expect, actually. Maybe it would be a future improvement)

Reply:
Added in